cpp/src/io/csv/reader_impl.cu

```diff
@@ -677,25 +677,29 @@ table_with_metadata read_csv(cudf::io::datasource* source,
   auto const& row_offsets = data_row_offsets.second;
 
   // Exclude the end-of-data row from number of rows with actual data
-  auto num_records        = std::max(row_offsets.size(), 1ul) - 1;
-  auto column_flags       = std::vector<column_parse::flags>();
-  auto column_names       = std::vector<std::string>();
-  auto num_actual_columns = static_cast<int32_t>(reader_opts.get_names().size());
-  auto num_active_columns = num_actual_columns;
-
-  // Check if the user gave us a list of column names
-  if (not reader_opts.get_names().empty()) {
-    column_flags.resize(reader_opts.get_names().size(),
-                        column_parse::enabled | column_parse::inferred);
-    column_names = reader_opts.get_names();
-  } else {
-    column_names = get_column_names(
-      header, parse_opts.view(), reader_opts.get_header(), reader_opts.get_prefix());
-
-    num_actual_columns = num_active_columns = column_names.size();
-
-    column_flags.resize(num_actual_columns, column_parse::enabled | column_parse::inferred);
-
+  auto num_records                   = std::max(row_offsets.size(), 1ul) - 1;
+  auto const unique_use_cols_indexes = std::set<int>(reader_opts.get_use_cols_indexes().cbegin(),
+                                                     reader_opts.get_use_cols_indexes().cend());
+
+  auto const detected_column_names =
+    get_column_names(header, parse_opts.view(), reader_opts.get_header(), reader_opts.get_prefix());
+  auto const opts_have_all_col_names =
+    not reader_opts.get_names().empty() and
+    (
+      // no data to detect (the number of) columns
+      detected_column_names.empty() or
+      // number of user specified names matches what is detected
+      reader_opts.get_names().size() == detected_column_names.size() or
```

Comment:
Probably a stupid edge case, but what happens if the user specifies the right number of column names but the names don't match (e.g. they're misspelled)? It looks like the current code will silently just return all columns in the file. Should there be some sort of error or log indicating that the user requested some nonexistent names?

Reply:
The names specified through `names` option (accessor `get_names()` used here) are meant to override the names in the file (if the header is present). This option does not mean column selection, names in `get_names` don't need to match the names in the file. 
Maybe you mixed it up with `use_cols_names` option (accessor `get_use_cols_names()`), which does column selection by their names.

Reply:
Oh I see, that makes sense. OK cool.